### ISSUE TITLE
[WebGPU] Error if any required feature is not available

### ISSUE THREAD

**Issue by IceSentry** [gfx-rs/wgpu#4194]
**Description**
When requesting POLYGON_MODE_LINE in webgpu it doesn't give an error despite it not working. Webgl correctly gives an error.

A similar issue also happen when using `PolygonMode::Line` without requesting the feature. In webgl it will give an error while in webgpu it gives no error but doesn't render correctly.

**Repro steps**
I discovered this with bevy. I tried making a simple reproduction but I couldn't figure out how to build a simple repro and build it for wasm, I kept getting unrelated errors, I was probably doing something wrong. Is there a simple template somewhere that I could use?

**Expected vs observed behavior**
Expected: All platform that do not support `POLYGON_MODE_LINE` to raise an error if the feature is requested
Observed: Some platform that do not support `POLYGON_MODE_LINE`  do not raise an error if the feature is requested

**Platform**
I was using windows, but I don't think this is OS related.
This was found with bevy 0.12-dev which is still on wgpu 0.16

**Comment by IceSentry** [comment]
@teoxoy I'm not sure that PR completely fixed the issue. Like I said, I'm not sure how to reproduce it outside of bevy, but that PR only fixed the primitive validation not the feature check validation. Was it just already fixed on main?

**Comment by cwfitzgerald** [comment]
Yeah we solved a different problem. The following code should error in the `None` case.

https://github.com/gfx-rs/wgpu/blob/25bc704e3515fdfd191bd414b6fa35e28f58d366/wgpu/src/backend/webgpu.rs#L1303-L1307
